iree/turbine/kernel/wave/templates/quantized_attention.py

```diff
@@ -168,13 +169,12 @@ def repeat(
             bias = tkw.select(mask, ZEROF, MIN_INF)
             x_j = x_j + bias
             x_j *= qk_scaling
-            x_j += fp8_offset
             m_j = tkw.max(x_j, partial_max, dim=N_KV)
             e_delta_max = tkw.exp2(partial_max - m_j)
             e_delta = tkw.exp2(x_j - m_j)
             e_init = partial_sum * e_delta_max
             d_j = tkw.sum(e_delta, e_init, dim=N_KV)
-            imm_f8 = low_precision_clamp(e_delta, fp8_max)
+            imm_f8 = low_precision_clamp(e_delta/fp8_offset, fp8_max)
```

Comment:
this is going to sound silly, but can we do:
```py
            imm_f8 = low_precision_clamp(e_delta * tkw.reciprocal(fp8_offset), fp8_max)
```

having direct div is really bad for performance

Reply:
the log2 addition makes sense, but it's not obvious to me how it allows us to make the  post scaling redundant?

Let fp8_offset = ATTENTION_SOFTMAX_MAX / F8_MAX
```py
fp8_offset = tkl.Register[B, N_Q, N_KV, tkl.f32](fp8_offset )
e_delta = tkw.exp2(x_j - m_j)
imm_f8 = low_precision_clamp(e_delta/fp8_offset, fp8_max)
```

-> 
```py
1/FP8_OFFSET_VAL
fp8_offset = tkl.Register[B, N_Q, N_KV, tkl.f32](1/FP8_OFFSET_VAL)
e_delta = tkw.exp2(x_j - m_j + fp8_offset)
imm_f8 = low_precision_clamp(e_delta/fp8_offset, fp8_max)
```

Reply:
Actually, seems like current version is doing `tkw.exp2(x_j - m_j + 1 / math.log2(F8_MAX/ATTENTION_SOFTMAX_MAX))`, so not too sure how we are doing that either actually 😅

Reply:
the fp8_dequant question is resolved offline:
```
we can remove the fp8_dequant because we are now applying /fp8_offset into e_delta -> d_j (denominator) as well as e_delta -> mma(numerator) => we no longer need to do fp8_dequant.
```

Reply:
> Actually, seems like current version is doing tkw.exp2(x_j - m_j + 1 / math.log2(F8_MAX/ATTENTION_SOFTMAX_MAX)), so not too sure how we are doing that either actually 😅

Resolved offline, it needs to be ` tkw.exp2(x_j - m_j + math.log2(F8_MAX/ATTENTION_SOFTMAX_MAX))`